# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [9]:
import pandas as pd
import numpy as np
from scipy import stats
import math

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [19]:
# Total number of callbacks

print ('Total callbacks =%d' % len(data[data['call'] == 1]))

# number of callbacks for black-sounding names
print('Black-sounding resume callbacks =%d' % sum(data[data.race=='b'].call))

# number of callbacks for white-sounding names
print('White-sounding resume callbacks =%d'% sum(data[data.race=='w'].call))

# number of NOcallbacks for black-sounding names
Bcall = data.loc[(data['race'] == 'b') & data['call'].isin([0])]
print('No callback for black-sounding resume =%d' % len(Bcall))

# number of NOcallbacks for white-sounding names
Wcall = data.loc[(data['race'] == 'w') & data['call'].isin([0])]
print('No callbacks for white-sounding resume =%d' % len(Wcall))

Total callbacks =392
Black-sounding resume callbacks =157
White-sounding resume callbacks =235
No callback for black-sounding resume =2278
No callbacks for white-sounding resume =2200


#### 1) What test is appropriate for this problem? Does CLT apply?

The paired two sample test would be the appropriate test to solve this problem, as two groups are assumed to be present. The sample size is 4870, which is greater than 30. The observations seem to be independent, as the selection of one does not affect the other. Assigning the 'b' and 'w' values was random. Therefore, the CLT seems to apply because of the reasons mentioned above.

#### 2) What are the null and alternate hypotheses?

Null hypothesis, Ho: black-sounding or white-sounding resumes does not have a significant effect to be called for an interview

Alternative hypothesis, Ha: black-sounding or white-sounding resumes has a significant effect to be called for an interview

#### 3) Compute margin of error, confidence interval, and p-value.

In [46]:
# margin of error = Z-score * standard deviation of sampling mean
# Let us test the hypothesis

mean_Bcall = 157/2435
mean_Wcall = 235/2435

print('mean of the black-sounding resume call back = %0.3f'% mean_Bcall) 
print('mean of the white-sounding resume call back = %0.3f'% mean_Wcall) 

# Finding the standard deviation of sample distribution of sampling mean for both resumes who got a callback

# Based on the null hypothesis, the probablity of getting a callback for both is assumed to be the same (Pb=Pw=P):

P=392/4870
print('probability of callback for any resume = %0.3f' % P)

# Standard deviation of the sampling mean

std_P = math.sqrt((2*(0.080)*(1-0.080)/2435))
print('standard deviation of sampling mean =%0.4f' % std_P)

# Z-score
Z_score=((0.097-0.064)-0)/(0.0078)
print('Z_score = %0.4f' % Z_score)

# Margin of error(E)
E =4.2308 * 0.0078
print('Margin of error = %0.4f' % E)

# The confidence interval for the Z-score of 4.2308 is:
CL=99.997
print('Confidence Interval = %0.4f' % CL)

# P-value:
P_value = 0.0023
print('P-value = %0.4f' % P_value)      

mean of the black-sounding resume call back = 0.064
mean of the white-sounding resume call back = 0.097
probability of callback for any resume = 0.080
standard deviation of sampling mean =0.0078
Z_score = 4.2308
Margin of error = 0.0330
Confidence Interval = 99.9970
P-value = 0.0023


#### 4)Statistical significance in the context of the original problem

From the above results, the probability of the mean for the black or white sounding resumes to get a callback (given the null hypothesis is true) is 0.23 % . The P-value is much less than our significance level of 5% (even 1%). The probability is small that the callback for the black or white-sounding happended was not by chance. We reject the null hypothesis and tend to accept the alternative hypothesis. Our findings are statistically significant. 


#### 5) Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would     you amend your analysis?

The race/name plays important factor in callback success based on the statistical analysis done above. The findings show that the statistic is significant that tells us the statistic is reliable. However, the significance difference can be large or small which will depend on the sample size.